acceptance/tests/pxp-module-puppet/run_puppet_result_failed.rb

```diff
@@ -0,0 +1,94 @@
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  target_identities = []
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment}" do
+    agents.each do |agent|
+      target_identities << "pcp://#{agent}/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'),
+               "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal(tmp_environment, action_result['environment'],
+                     "Result of pxp-module-puppet run on #{agent} should run with the "\
+                     "#{tmp_environment} environment")
+        assert_equal('failed', action_result['status'],
+                     "Result of pxp-module-puppet run on #{agent} should be 'failed'. was: "\
+                     "'#{action_result["status"]}'")
+      end
+    end
+  end
+
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment2}" do
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment2,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'),
+               "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal(tmp_environment2, action_result['environment'],
+                     "Result of pxp-module-puppet run on #{agent} should run with the "\
+                     "#{tmp_environment2} environment")
+        assert_equal('failed', action_result['status'],
```

Comment:
It may be valuable to assert the exit code in this case; but when I had a quick look last week, I noticed it was possibly coming out wrong - I'll raise a separate ticket to look into that instead of dragging out this ticket longer